include/laser_filters/angular_interval_filter.h

```diff
@@ -0,0 +1,223 @@
+
+#ifndef LASER_SCAN_ANGULAR_INTERVAL_FILTER_H
+#define LASER_SCAN_ANGULAR_INTERVAL_FILTER_H
+/**
+\author Ulas Sureyya Bingol
+@b ScanAngularIntervalFilter takes input scans and filters the readings within the ranges of interval sets.
+**/
+
+
+#include "filters/filter_base.h"
+#include "sensor_msgs/LaserScan.h"
+#include <laser_filters/AngularIntervalFilterConfig.h>
+#include <dynamic_reconfigure/server.h>
+
+namespace laser_filters
+{
+
+class LaserScanAngularIntervalFilter : public filters::FilterBase<sensor_msgs::LaserScan>
+{
+public:
+  std::string interval_sets_;
+  std::vector<std::vector<double>> intervals_;
+  std::shared_ptr<dynamic_reconfigure::Server<laser_filters::AngularIntervalFilterConfig>> dyn_server_;
+  AngularIntervalFilterConfig param_config;
+  boost::recursive_mutex own_mutex_;
+
+  bool enable_ = true;
+  bool got_intervals_ = false;
+  bool configure()
+  {
+    ros::NodeHandle private_nh("~" + getName());
+    dyn_server_.reset(new dynamic_reconfigure::Server<laser_filters::AngularIntervalFilterConfig>(own_mutex_, private_nh));
+    dynamic_reconfigure::Server<laser_filters::AngularIntervalFilterConfig>::CallbackType f;
+    f = boost::bind(&laser_filters::LaserScanAngularIntervalFilter::reconfigureCB, this, _1, _2);
+    dyn_server_->setCallback(f);
+
+
+    if (!filters::FilterBase<sensor_msgs::LaserScan>::getParam(std::string("interval_sets"), interval_sets_))
+    {
+      ROS_WARN("AngularIntervalFilter was not given interval sets.\n");
+    }
+    else
+    {
+      param_config.interval_sets = interval_sets_;
+      dyn_server_->updateConfig(param_config);
+    }
+    return true;
+  }
+
+  virtual ~LaserScanAngularIntervalFilter()
+  {
+
+  }
+
+  bool update(const sensor_msgs::LaserScan& input_scan, sensor_msgs::LaserScan& filtered_scan)
+  {
+    filtered_scan = input_scan;
+    double replacement_value = std::numeric_limits<double>::quiet_NaN();
+    double current_angle = input_scan.angle_min;
+    if(got_intervals_ && enable_)
+    {
+      for (unsigned int i = 0; i < input_scan.ranges.size(); i++) // Need to check ever reading in the current scan
+      {
+        current_angle = input_scan.angle_min + i* input_scan.angle_increment;
+        for(auto interval : intervals_)
+        {
+          if(current_angle < interval[1]  && current_angle > interval[0])
+            filtered_scan.ranges[i] = replacement_value;
+        }
+      }
+    }
+    return true;
+  }
+  
+  void reconfigureCB(AngularIntervalFilterConfig& config, uint32_t level)
+  {
+  	interval_sets_ = config.interval_sets; // th11,th12;th21,th22;th31,th32
+  	enable_ = config.enable;
+
+    std::string text = interval_sets_;
+    std::string parse_error;
+    if(parseIntervalString(text, parse_error, intervals_))
+    {
+      ROS_INFO("Got intervals %s", parse_error.c_str());
+      got_intervals_ = true;
+    }
+    else
+    {
+      got_intervals_ = false;
+      ROS_WARN("Failed to calculate intervals: %s", parse_error.c_str());
```

Comment:
I think it would be cleaner here to throw an exception on an error immediately, rather than printing out a warning. (This is what polygon filter does on a bad reconfigure parameter: https://github.com/ros-perception/laser_filters/blob/353392eb6d154312dd47c4d477f6324ae5782cdc/src/polygon_filter.cpp#L94 )